### PR TITLE
test: add a11y tests

### PR DESCRIPTION
**Pull request by mxschmitt**
Motivation: As per Microsoft Accessibility Standards we need to setup automation for running a11y tests on our website. Starting with `/` and `/intro` for now. They don't provide a GitHub Actions integration so we need to use Azure Pipelines in order to report the results back to their platform.

### CODE REVIEW

File: playwright.config.ts [hunk 1]
```diff
@@ -63,8 +62,8 @@ const config: PlaywrightTestConfig = {
   webServer: {
     command: 'npx http-server -p 3000 build/',
     port: 3000,
-    reuseExistingServer: !process.env.CI,
+    reuseExistingServer: true
```
Comment (dgozman) [inline playwright.config.ts:65]: Why this change?

File: tests/a11y.spec.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (dgozman) [inline tests/a11y.spec.ts:1]: We use a different copyright everywhere. The same applies to the next file.

File: .azure-pipelines/publish.yml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+          inputs:
+            version: '18.x'
+          displayName: 'Install Node.js'
+        - script: npm ci
+          displayName: Install
+        - script: npm run build
+          displayName: Build
+        - script: npx playwright install --with-deps
+          displayName: 'Install Playwright Browsers'
+        - script: npx playwright test
```
Comment (dgozman) [inline .azure-pipelines/publish.yml:45]: Perhaps only run accessibility tests here? I am not sure how the integration works with the insights tool.

File: tests/a11y.spec.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+      await page.goto(url);
+      const accessibilityScanResults = await new AxeBuilder({ page })
+        // This withTags directive restricts Axe to only run tests that detect known violations of
+        // WCAG 2.1 A and AA rules (similar to what Accessibility Insights reports). If you omit
+        // this, Axe will additionally run several "best practice" rules, which are good ideas to
+        // check for periodically but may report false positives in certain edge cases.
+        .withTags(['wcag2a', 'wcag2aa', 'wcag21a', 'wcag21aa'])
+        .analyze();
+
+      await exportAxeAsSarifTestResult('index-h1-element.sarif', accessibilityScanResults);
```
Comment (dgozman) [inline tests/a11y.spec.ts:19]:
```suggestion
      await exportAxeAsSarifTestResult(`${slugify(url)}.sarif`, accessibilityScanResults);
```

File: tests/export-to-sarif.ts [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+//
+// Exporting axe results as .sarif files lets our Azure Pipelines build results page show a nice visualization
+// of any accessibility failures we find using the Sarif Results Viewer Tab extension
+// (https://marketplace.visualstudio.com/items?itemName=sariftools.sarif-viewer-build-tab)
+export async function exportAxeAsSarifTestResult(sarifFileName: string, axeResults: Axe.AxeResults): Promise<void> {
+    // We use the axe-sarif-converter package for the conversion step, then write the results
+    // to a file that we'll be publishing from a CI build step in azure-pipelines.yml
+    const sarifResults = convertAxeToSarif(axeResults);
+
+    const sarifResultFile = test.info().outputPath(sarifFileName);
```
Comment (dgozman) [inline tests/export-to-sarif.ts:18]: Seems strange to output to `test-results`. Perhaps we should at least attach it? Although, as long as it works for accessibility insights integration...

**Review by dgozman — APPROVED** [review]

File: playwright.config.ts (hunk 1 above)
Comment (mxschmitt) [reply playwright.config.ts]: Reverted.
